From 5a3151f790d4ed6608335a2182bfc0d2002855a0 Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Tue, 23 Jan 2018 23:12:23 +0800
Subject: [PATCH 1/3] ARM: dts: imx6sx: add pu power domain support

commit 7a485590960dff9f4cac42d9a197c7f9fbfae376 upstream

Add PU power domain support, GPU is the only
module inside PU power domain, and PU power
is supplied by LDO_SOC.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Acked-by: Dong Aisheng <aisheng.dong@nxp.com>
Signed-off-by: Shawn Guo <shawnguo@kernel.org>
[Quanyang: add modification from "gpc 1" to "pd_pu"]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 arch/arm/boot/dts/imx6sx.dtsi |    9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/imx6sx.dtsi b/arch/arm/boot/dts/imx6sx.dtsi
index 8e1f520..c0b80a3 100644
--- a/arch/arm/boot/dts/imx6sx.dtsi
+++ b/arch/arm/boot/dts/imx6sx.dtsi
@@ -270,7 +270,7 @@
 				"gpu3d_shader_clk";
 			resets = <&src 0>;
 			reset-names = "gpu3d";
-			power-domains = <&gpc 1>;
+			power-domains = <&pd_pu>;
 		};
 
 		gpmi: gpmi-nand@01806000{
@@ -884,6 +884,13 @@
 					#address-cells = <1>;
 					#size-cells = <0>;
 
+					pd_pu: power-domain@1 {
+						reg = <1>;
+						#power-domain-cells = <0>;
+						power-supply = <&reg_soc>;
+						clocks = <&clks IMX6SX_CLK_GPU>;
+					};
+
 					pd_pci: power-domain@3 {
 						reg = <3>;
 						#power-domain-cells = <0>;
-- 
1.7.9.5

